andela/knights-bn-backend
# #170947658 Signup Validation

#### What does this PR do?
 
- This PR will validate data inserted by a user while signing up

#### Description of Task to be completed?

- check if the data inserted by the user are corrected

- write the test for those validations

#### How should this be manually tested?
- open your preferred code editor 
- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo.
- run `cd knights-bn-backend` to go to the project repo.
- run `code .` to open the project
- run `git checkout `ft-signup-validation-170947658` to switch to this branch.
- run `npm install` to install dependencies.
- create `.env` file and populate it, refer to `env.example` to see what fill in .env
- create `database` for both `development` and `test`
- fill the names of the databases you created under `.env`
- run `npm run createTables` to create Tables.
- run `npm run start` to start the server.
- Open [postman](https://www.postman.com/) use action as `POST` then use this URL `localhost:4000/api/v1/auth/signup` a
- Required inputs are: 
    
           `firstName | lastName | gender | passportNumber | email | password`

- Here is an `example` of a valid request body :
`{
	"firstName": "Niyonsenga",
	"lastName": "Eric",
	"gender": "Male",
	"PassportNumber": "123456789",
	"email": "example@gmail.com ",
	"password": "Niyonkuru@1"
}`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[https://www.pivotaltracker.com/story/show/170947658](https://www.pivotaltracker.com/story/show/170947658)

#### Screenshots

       Invalid first name

<img width="1167" alt="invalid firstName" src="https://user-images.githubusercontent.com/48021034/74253976-7c588d80-4cf8-11ea-90d8-2ef0bf17585b.png">

       Invalid last name

<img width="1166" alt="Invalid lastName" src="https://user-images.githubusercontent.com/48021034/74254529-3d770780-4cf9-11ea-9c77-8082fb8c09ce.png">

      Invalid gender

<img width="1172" alt="invalid gender" src="https://user-images.githubusercontent.com/48021034/74254575-4c5dba00-4cf9-11ea-81ef-3daf9f9b6f97.png">


      Invalid Email

<img width="1173" alt="invalid email" src="https://user-images.githubusercontent.com/48021034/74254599-57184f00-4cf9-11ea-943b-8e3889fab50b.png">


      weak password

<img width="1178" alt="weak password" src="https://user-images.githubusercontent.com/48021034/74254643-65666b00-4cf9-11ea-8a39-c480e519815a.png">


       Existing user

<img width="1167" alt="Screen Shot 2020-02-11 at 18 08 49" src="https://user-images.githubusercontent.com/48021034/74254778-99419080-4cf9-11ea-808b-4a75a2ad7307.png">


      Existing passport number

<img width="1167" alt="passport exists" src="https://user-images.githubusercontent.com/48021034/74254822-aa8a9d00-4cf9-11ea-9e23-39b2830ecf0a.png">



      validation Tests

<img width="1170" alt="Screen Shot 2020-02-11 at 18 18 28" src="https://user-images.githubusercontent.com/48021034/74255720-f2f68a80-4cfa-11ea-8155-9ff5deb4f827.png">
